Log warnings after updating ESPHome to V2025.11.10

Although the wMBus-Gateway code compiles and uploads OK, there are now a number of warnings in the log when compiling against the latest version of ESPHome (V2025.11.10).

I have included an abbreviated log file with the relevant warning messages below:

```.
.
.
Compiling .pioenvs/wmbusgateway/src/esphome/components/wmbus_gateway/display_manager.cpp.o
Compiling .pioenvs/wmbusgateway/src/esphome/components/wmbus_gateway/screen.cpp.o
In file included from src/esphome/components/wmbus_gateway/display_manager.h:17,
                 from src/esphome/components/wmbus_gateway/display_manager.cpp:1:
src/esphome/components/wmbus_gateway/screen_sensor.h: In member function 'void esphome::wmbus_gateway::SensorScreen::render_value(esphome::display::Display&)':
src/esphome/components/wmbus_gateway/screen_sensor.h:46:101: warning: 'std::string esphome::EntityBase_UnitOfMeasurement::get_unit_of_measurement()' is deprecated: Use get_unit_of_measurement_ref() instead for better performance (avoids string copy). Will be removed in ESPHome 2026.5.0 [-Wdeprecated-declarations]
   46 |                         data = str_truncate(data, std::min(pos, 11 - sensor->get_unit_of_measurement().length()));
      |                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from src/esphome/components/binary_sensor/binary_sensor.h:3,
                 from src/esphome/components/wmbus_gateway/display_manager.h:6:
src/esphome/core/entity_base.h:183:15: note: declared here
  183 |   std::string get_unit_of_measurement();
      |               ^~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wmbus_gateway/screen_sensor.h:48:60: warning: 'std::string esphome::EntityBase_UnitOfMeasurement::get_unit_of_measurement()' is deprecated: Use get_unit_of_measurement_ref() instead for better performance (avoids string copy). Will be removed in ESPHome 2026.5.0 [-Wdeprecated-declarations]
   48 |                     data += sensor->get_unit_of_measurement();
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/esphome/core/entity_base.h:183:15: note: declared here
  183 |   std::string get_unit_of_measurement();
      |               ^~~~~~~~~~~~~~~~~~~~~~~
Compiling .pioenvs/wmbusgateway/src/esphome/components/wmbus_meter/base_sensor.cpp.o
Compiling .pioenvs/wmbusgateway/src/esphome/components/wmbus_meter/sensor.cpp.o
.
.
.
Compiling .pioenvs/wmbusgateway/lib513/libsodium/randombytes/randombytes.o
Compiling .pioenvs/wmbusgateway/lib513/libsodium/randombytes/sysrandom/randombytes_sysrandom.o
In file included from src/esphome/components/wmbus_gateway/display_manager.h:17,
                 from src/esphome.h:105,
                 from src/main.cpp:3:
src/esphome/components/wmbus_gateway/screen_sensor.h: In member function 'void esphome::wmbus_gateway::SensorScreen::render_value(esphome::display::Display&)':
src/esphome/components/wmbus_gateway/screen_sensor.h:46:101: warning: 'std::string esphome::EntityBase_UnitOfMeasurement::get_unit_of_measurement()' is deprecated: Use get_unit_of_measurement_ref() instead for better performance (avoids string copy). Will be removed in ESPHome 2026.5.0 [-Wdeprecated-declarations]
   46 |                         data = str_truncate(data, std::min(pos, 11 - sensor->get_unit_of_measurement().length()));
      |                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from src/esphome/components/binary_sensor/binary_sensor.h:3,
                 from src/esphome/core/application.h:30,
                 from src/esphome/components/api/api_frame_helper.h:13,
                 from src/esphome/components/api/api_connection.h:5,
                 from src/esphome.h:3:
src/esphome/core/entity_base.h:183:15: note: declared here
  183 |   std::string get_unit_of_measurement();
      |               ^~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wmbus_gateway/screen_sensor.h:48:60: warning: 'std::string esphome::EntityBase_UnitOfMeasurement::get_unit_of_measurement()' is deprecated: Use get_unit_of_measurement_ref() instead for better performance (avoids string copy). Will be removed in ESPHome 2026.5.0 [-Wdeprecated-declarations]
   48 |                     data += sensor->get_unit_of_measurement();
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/esphome/core/entity_base.h:183:15: note: declared here
  183 |   std::string get_unit_of_measurement();
      |               ^~~~~~~~~~~~~~~~~~~~~~~
Compiling .pioenvs/wmbusgateway/lib513/libsodium/sodium/codecs.o
Compiling .pioenvs/wmbusgateway/lib513/libsodium/sodium/core.o
.
.
.
```

--- a/components/wmbus_gateway/screen_sensor.h
+++ b/components/wmbus_gateway/screen_sensor.h
@@ -43,9 +43,9 @@ namespace esphome
                     {
                         if (sensor->get_accuracy_decimals() > 0)
                             pos += 1 + sensor->get_accuracy_decimals();
-                        data = str_truncate(data, std::min(pos, 11 - sensor->get_unit_of_measurement().length()));
+                        data = str_truncate(data, std::min(pos, 11 - sensor->get_unit_of_measurement_ref().size()));
                     }
-                    data += sensor->get_unit_of_measurement();
+                    data += sensor->get_unit_of_measurement_ref();
                 }
                 it.print(it.get_width(),
                          it.get_height() / 2,

--- a/factory.yaml
+++ b/factory.yaml
@@ -1,5 +1,5 @@
-packages: !include
-  file: packages/wmbus_gateway.yaml
+packages:
+  wmbus_gateway: !include packages/wmbus_gateway.yaml
 
 improv_serial:
 

